### PR TITLE
fix: makes test archive name unique to support parallel testing

### LINKED ISSUE #304: RemoteExtensionLoader swallows IOException
##### Issue Overview
When debugging parallel testing, errors occur that get swallowed by RemoteExtensionLoader

##### Expected Behaviour
Exceptions should be printed out

##### Current Behaviour
Exceptions are silently ignored

##### Steps To Reproduce
Parallel test Java EE 7 samples with Payara `mvn -T4 verify`

### LINKED ISSUE #305: Parallel testing of JAR deployments fails due to hardcoded "test.war"
##### Issue Overview
`test.war` is hardcoded when using JAR deployment.
This causes multiple failed deployments of said `test.war` thus failing parallel testing

Tell us briefly what the problem is about.

##### Expected Behaviour
Parallel tests succeed

##### Current Behaviour
Parallel tests with remote server fails due to duplicate deployment name `test.war`

##### Steps To Reproduce
Test Java EE 7 samples against Payara `mvn -T3 verify`

### PR DESCRIPTION
**Pull request by lprimak**
Feature:
- Support parallel testing in JAR deployments by making generated WAR name UUID rather than "test.war"

Bugs:
- catch IOException in remote extension loader and throw appropriate error rather than swallowing the exception

Fixes #304
Fixes #305

### CODE REVIEW

File: protocols/servlet-jakarta/src/main/java/org/jboss/arquillian/protocol/servlet5/v_5/ServletProtocolDeploymentPackager.java [hunk 1]
```diff
@@ -92,7 +93,7 @@
     private Archive<?> handleArchive(JavaArchive applicationArchive, Collection<Archive<?>> auxiliaryArchives,
         JavaArchive protocol, Processor processor) {
         return handleArchive(
-            ShrinkWrap.create(WebArchive.class, "test.war")
+            ShrinkWrap.create(WebArchive.class, UUID.randomUUID().toString() + ".war")
```
Comment (bartoszmajsak) [inline protocols/servlet-jakarta/src/main/java/org/jboss/arquillian/protocol/servlet5/v_5/ServletProtocolDeploymentPackager.java:96]:
```suggestion
            ShrinkWrap.create(WebArchive.class, "test-" + UUID.randomUUID().toString() + ".war")
```

I would keep `test-` prefix at least. This way it's more obvious what this archive could be.

File: protocols/servlet/src/main/java/org/jboss/arquillian/protocol/servlet/v_3/ServletProtocolDeploymentPackager.java [hunk 2]
```diff
@@ -91,7 +92,7 @@
     private Archive<?> handleArchive(JavaArchive applicationArchive, Collection<Archive<?>> auxiliaryArchives,
         JavaArchive protocol, Processor processor) {
         return handleArchive(
-            ShrinkWrap.create(WebArchive.class, "test.war")
+            ShrinkWrap.create(WebArchive.class, UUID.randomUUID().toString() + ".war")
```
Comment (bartoszmajsak) [inline protocols/servlet/src/main/java/org/jboss/arquillian/protocol/servlet/v_3/ServletProtocolDeploymentPackager.java:95]:
```suggestion
            ShrinkWrap.create(WebArchive.class, "test-" + UUID.randomUUID().toString() + ".war")
```

Same reason as above.

**Review by bartoszmajsak — CHANGES_REQUESTED** [review]
Thank you very much for this improvement! Much appreciated. 

I just have two minor suggestions and we can merge it.

Please update the branch to the latest `master` or let me edit this PR so I can take care of it :)

**Comment by lprimak** [comment]
@bartoszmajsak all done!

**Review by bartoszmajsak — APPROVED** [review]

**Comment by bartoszmajsak** [comment]
Awesome, thank you!

**Comment by arjantijms** [comment]
Hey nice to see this finally being done! @lprimak is this from the issue I created years ago, or did the same problem just happen to bubble up again?

**Comment by lprimak** [comment]
Probably the same issue you had, but independently noticed :)

**Pull request merged**